qiskit_ibm_provider/ibm_backend_service.py

```diff
@@ -238,20 +260,45 @@ def jobs(
         if instance:
             api_filter["provider"] = instance
         # Retrieve all requested jobs.
-        job_responses = self._get_jobs(
-            api_filter=api_filter, limit=limit, skip=skip, descending=descending
+        filter_by_status = (
+            status
+            and status not in ["pending", "completed"]
+            and status in [status.name for status in JobStatus]
         )
         job_list = []
-        for job_info in job_responses:
-            # TODO filter by status
-            job = self._restore_circuit_job(job_info, raise_error=False)
-            if job is None:
-                logger.warning(
-                    'Discarding job "%s" because it contains invalid data.',
-                    job_info.get("job_id", ""),
-                )
-                continue
-            job_list.append(job)
+        original_limit = limit
+        while True:
+            job_responses = self._get_jobs(
+                api_filter=api_filter, limit=limit, skip=skip, descending=descending
+            )
+            if len(job_responses) == 0:
+                break
+            different_status_jobs = 0
+            for job_info in job_responses:
+                if filter_by_status:
+                    job_status = api_status_to_job_status(
+                        job_info["state"]["status"].upper()
+                    ).name
+                    if (isinstance(status, str) and job_status != status) or (
+                        isinstance(status, list) and job_status not in status
+                    ):
+                        continue
+                    different_status_jobs += 1
+                job = self._restore_circuit_job(job_info, raise_error=False)
+                if job is None:
+                    logger.warning(
+                        'Discarding job "%s" because it contains invalid data.',
+                        job_info.get("job_id", ""),
+                    )
+                    continue
+                job_list.append(job)
+                if (
+                    len(job_list) == original_limit
+                    or different_status_jobs == original_limit
+                ):
+                    return job_list
+            limit += limit
```

Comment:
I'm assuming you need to increment limit here because that's how the API works? e.g. if there are 40 jobs, and you specify `skip=20, limit=20`, it won't actually return any (because those 20 are "skipped")? Therefore you need to do `skip=20, limit=40`?

Reply:
you're right, I don't need to increment the limit here `limit=20, skip=20` will still return 20 jobs

fixed